### PR TITLE
[0.2.0] pick block/entity hooks

### PR DESCRIPTION
**Pull request by asiekierka**

### CODE REVIEW

**Comment by asiekierka** [comment]
It is not unlikely that I will move the block/entity interfaces to more specific mixins in a later update, as to work with other non-Fabric mods hooking those areas. :thinking:

**Comment by JamiesWhiteShirt** [comment]
I think using the same class name in different packages is going to be a bit of a pain. It's fine when you can statically determine that something is incorrect (incompatible types) but in this case you could easily implement the wrong interface on the right class without noticing.

**Comment by asiekierka** [comment]
@JamiesWhiteShirt Fair. Do you think we should suffix those interfaces with Block/Entity/etc.? This might get confusing with the classes themselves.

File: src/main/java/net/fabricmc/fabric/impl/FabricAPIClientInitializer.java [hunk 1]
```diff
@@ -30,6 +38,26 @@ public void onInitializeClient() {
...
+				BlockPos pos = ((BlockHitResult) result).getBlockPos();
+				BlockState state = view.getBlockState(pos);
+
+				if (state.getBlock() instanceof ContextSensitivePickable) {
+					container.setStack(((ContextSensitivePickable) state.getBlock()).getPickStack(state, view, pos, player, result));
+				}
+			} else if (result instanceof EntityHitResult) {
+				Entity entity = ((EntityHitResult) result).getEntity();
+
+				if (entity instanceof net.fabricmc.fabric.api.entity.ContextSensitivePickable) {
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/fabric/impl/FabricAPIClientInitializer.java]: Yeah, I agree these should really have 2 different names.

**Pull request merged**